src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -79,12 +79,15 @@ class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
 
   fetchMetrics() {
     console.log('Fetching metrics...');
-    axios
-      .get(`/api/namespaces/${this.props.namespace}/services/${this.props.service}/metrics?range=${this.state.range}`)
-      // .get(`http://127.0.0.1:8000/api/namespaces/${this.props.namespace}/services/${this.props.service}/metrics?range=${this.state.range}`)
+    API.newRequest(
```

Comment:
I still prefer `API.GetServiceMetrics(this.props.namespace, this.props.service, this.state.range)`
Config-based endpoints is more error prone. The compiler will never tell you if you're trying to reach an endpoint that doesn't exist

Reply:
So...We'll need a method for each endpoint. What do you think about pass the query params like range inside a json to let axios work with them?

like  API.GetServiceMetrics(this.props.namespace, this.props.service, {range :this.state.range})
with this we can add more queryparams in future without change the method.

Reply:
+1 :-)

Reply:
Changed @jotak, review please